nntrainer/layers/cl_layers/swiglu_cl.cpp

```
@@ -116,91 +116,66 @@ void SwiGLULayerCl::swigluProcess(Tensor const &in1, Tensor const &in2,
-      if (!map_result) {
-        ml_loge("Failed to map svm");
-        break;
-      }
-
-      bool set_svm_result = true;
-      set_svm_result &= kernel_swiglu_ptr->SetKernelSVMArguments(0, matAdata);
-      set_svm_result &= kernel_swiglu_ptr->SetKernelSVMArguments(1, vecXdata);
-      set_svm_result &= kernel_swiglu_ptr->SetKernelSVMArguments(2, vecYdata);
-      if (!set_svm_result) {
-        ml_loge("Failed to set svm");
-        break;
-      }
-    }
+  if (!use_svm) {
+    bool write_result = true;
+    auto &clbuffInstance = ClBufferManager::Global();
 
-    // NOTE(mwlasiuk) : local size can not be larger than global
-    const int32_t desired_local = 64;
-    const bool can_use_desired = dim >= desired_local;
-    const int32_t chosen_local = can_use_desired ? desired_local : dim;
+    write_result &= clbuffInstance.getInBufferA()->WriteDataRegion(
+      cl_context->command_queue_inst_, dim * sizeof(float), matAdata);
+    write_result &= clbuffInstance.getInBufferB()->WriteDataRegion(
+      cl_context->command_queue_inst_, dim * sizeof(float), vecXdata);
+    if (!write_result) {
+      return;
+    }
 
-    const int work_groups_count[3] = {dim, 1, 1};
-    /// @todo: create a group size by device & input
-    const int work_group_size[3] = {chosen_local, 1, 1}; // test-value
+    auto bufferInA = clbuffInstance.getInBufferA()->GetBuffer();
+    auto bufferInB = clbuffInstance.getInBufferB()->GetBuffer();
+    auto bufferOutA = clbuffInstance.getOutBufferA()->GetBuffer();
 
-    if (!global_cl_context->command_queue_inst_.DispatchCommand(
-          kernel_swiglu_ptr, work_groups_count, work_group_size)) {
-      ml_loge("Failed to run");
-      break;
+    bool set_result = true;
+    set_result &= kernel->SetKernelArguments(0, &bufferInA, sizeof(cl_mem));
+    set_result &= kernel->SetKernelArguments(1, &bufferInB, sizeof(cl_mem));
+    set_result &= kernel->SetKernelArguments(2, &bufferOutA, sizeof(cl_mem));
+    if (!set_result) {
```

missing ml_loge message

---

I think that we should change SetKernelArguments to throw in case of error instead of return bool in some next PR. This way we will avoid need of checking ret value all the time